scripts/create-kubestellar-demo-env.sh

```diff
@@ -134,14 +97,85 @@ checking_cluster() {
             
         else
             echo "$1 not found. Trying again..."
-            sleep 5
+            sleep 20
         fi
 
     done
 }
+##########################################
+
+echo -e "\nStarting environment clean up..."
+echo -e "Starting cluster clean up..."
+
+yq -i 'del(.preferences)' ${KUBECONFIG:-$HOME/.kube/config} 
+
+cluster_clean_up "kind delete cluster --name kubeflex" &
+cluster_clean_up "kind delete cluster --name cluster1" &
+cluster_clean_up "kind delete cluster --name cluster2" &
+wait
+echo -e "Cluster space clean up has been completed"
+
+echo -e "\nStarting context clean up..."
+context_clean_up
+echo "Context space clean up completed"
+
+echo -e "\nStarting the process to install KubeStellar core: kind-kubeflex..."
+clusters=(cluster1 cluster2)
+for cluster in "${clusters[@]}"; do
+   (
+     echo -e "Creating cluster ${cluster}..."
+     kind create cluster --name "${cluster}" >/dev/null 2>&1 &&
+     kubectl config rename-context "kind-${cluster}" "${cluster}" >/dev/null 2>&1
+     echo -e "${cluster} creation and context setup complete"
+   ) &
+done
+wait 
+
+echo -e "Creating KubeFlex cluster with SSL Passthrough"
+curl -s https://raw.githubusercontent.com/kubestellar/kubestellar/v${kubestellar_version}/scripts/create-kind-cluster-with-SSL-passthrough.sh | bash -s -- --name kubeflex --port 9443 
+echo -e "Completed KubeFlex cluster with SSL Passthrough"
+
+echo -e "\nPulling container images local..."
+images=("ghcr.io/loft-sh/vcluster:0.16.4"
+        "rancher/k3s:v1.27.2-k3s1"
+        "quay.io/open-cluster-management/registration-operator:v0.13.2"
+        "docker.io/bitnami/postgresql:16.0.0-debian-11-r13")
+
+for image in "${images[@]}"; do
+    (
+        docker pull "$image" && kind load docker-image "$image" --name kubeflex
+    ) &
+done
+
```

Comment:
FYI, I made a modified version of this script with an added `wait` here, to get all the prefetching done before starting the chart instantiation. After warming up the caches, in two consecutive trials I got the following timings.

```
scripts/create-kubestellar-demo-env.sh  6.40s user 6.90s system 4% cpu 5:22.06 total
scripts/create-kubestellar-demo-env.sh  6.38s user 6.74s system 3% cpu 5:45.40 total
```

Without the `wait`, the timings are a bit quicker:

```
scripts/create-kubestellar-demo-env.sh  6.18s user 6.71s system 4% cpu 5:10.15 total
scripts/create-kubestellar-demo-env.sh  6.33s user 6.98s system 4% cpu 5:25.90 total
```

Reply:
I wouldn't wait, the ks-core is loading many other container images that takes about 3 to 4 minutes. This allows plenty of time for the 4 pre-fetches to complete. If you wait for the pre-fetch to complete, then your just delaying ks-core which does not depend immediately on any of the pre-fetches

Reply:
Indeed. I thought it worth documenting the actual results.